R/dm.R

```diff
@@ -70,11 +70,34 @@ new_dm <- function(src, tables, data_model) {
   stopifnot(datamodelr::is.data_model(data_model))
   src <- src_from_src_or_con(src)
 
+  columns <- data_model$columns
+
+  keys <- columns %>%
+    select(column, table, key) %>%
+    filter(key > 0) %>%
+    select(-key)
+
+  if (is.null(data_model$references)) {
+    references <- data.frame(
+      table = character(),
+      column = character(),
+      ref = character(),
+      ref_col = character(),
+      stringsAsFactors = FALSE
+    )
+  } else {
+    references <-
+      data_model$references %>%
+      select(table, column, ref, ref_col)
+  }
+
   structure(
     list(
       src = src,
       tables = tables,
-      data_model = data_model
+      data_model_tables = data_model$tables,
+      data_model_keys = keys,
+      data_model_references = references
```

Comment:
Maybe we talked about this already, but it does seem very complicated to always have to change between `data_model` and the three parts needed for our class. For now it seems good though.

Only thing to consider is if we want to store the data types as well, since we seem to need it for MSSQL, but actually we can just get these easily enough from the tables themselves.